pennylane/transforms/qcut.py

```diff
@@ -1658,3 +1662,392 @@ def _infer_probed_cuts(
+    <networkx.classes.multidigraph.MultiDiGraph at 0x7f7251ac1220>
+
+    And visualize the cut by converting back to a tape:
+
+    >>> print(qml.transforms.qcut.graph_to_tape(cut_graph).draw())
+     0: ──RX(0.432)──────╭C──┤ ⟨Z⟩
+     a: ──RY(0.543)──//──╰X──┤
+    """
+    cut_graph = graph.copy()
+
+    for op0, op1, wire_key in cut_edges:
+        # Get info:
+        order = cut_graph.nodes[op0]["order"] + 1
+        wire = cut_graph.edges[(op0, op1, wire_key)]["wire"]
+        # Apply cut:
+        cut_graph.remove_edge(op0, op1, wire_key)
+        # Increment order for all subsequent gates:
+        for op, o in cut_graph.nodes(data="order"):
+            if o >= order:
+                cut_graph.nodes[op]["order"] += 1
+        # Add WireCut
+        wire_cut = WireCut(wires=wire)
+        cut_graph.add_node(wire_cut, order=order)
+        cut_graph.add_edge(op0, wire_cut, wire=wire)
+        cut_graph.add_edge(wire_cut, op1, wire=wire)
+
+    return cut_graph
+
+
+def _remove_existing_cuts(graph: MultiDiGraph) -> MultiDiGraph:
+    """Removes all existing, manually or automatically placed, cuts from a circuit graph, be it
+    ``WireCut``s or ``MeasureNode``-``PrepareNode`` pairs.
+
+    Args:
+        graph (MultiDiGraph): The original (tape-converted) graph to be cut.
+
+    Returns:
+        (MultiDiGraph): Copy of the input graph with all its existing cuts removed.
+    """
+    uncut_graph = graph.copy()
+    for op in list(graph.nodes):
+        if isinstance(op, WireCut):
+            uncut_graph.remove_node(op)
+        elif isinstance(op, MeasureNode):
+            for op1 in graph.neighbors(op):
+                if isinstance(op1, PrepareNode):
+                    uncut_graph.remove_node(op)
+                    uncut_graph.remove_node(op1)
+
+    if len([n for n in uncut_graph.nodes if isinstance(n, (MeasureNode, PrepareNode))]) > 0:
+        warnings.warn(
+            "The circuit contains `MeasureNode`/`PrepareNode` that is not paired up correctly. "
+            "Please check.",
+            UserWarning,
+        )
+    return uncut_graph
+
+
+def find_and_place_cuts(
+    graph: MultiDiGraph, cut_method: Callable = kahypar_cut, replace_wire_cuts=False, **kwargs
+) -> MultiDiGraph:
+    """Automatically finds and places optimal :class:`~.WireCut` nodes into a given tape-converted graph
+    using a customizable graph partitioning function. Preserves existing placed cuts.
+
+    Args:
+        graph (MultiDiGraph): The original (tape-converted) graph to be cut.
+        cut_method (Callable): A graph partitioning function that takes an input graph and returns
+            a list of edges to be cut based on a given set of constraints and objective. Defaults
+            to :func:`kahypar_cut` which requires KaHyPar to be installed using
+            ``pip install kahypar`` for Linux and Mac users or visiting the
+            instructions `here <https://kahypar.org/#the-python-interface>`__ to compile from
```

Comment:
```suggestion
            instructions `here <https://kahypar.org>`__ to compile from
```
I realized the python-based instructions look linux specific.